dspace-api/src/main/java/org/dspace/app/canvasdimension/IIIFCanvasDimensionServiceImpl.java

```diff
@@ -0,0 +1,219 @@
+            if (!isQuiet) {
+                System.out.println("Updated canvas metadata for item: " + item.getID());
+            }
+        }
+        return done;
+
+    }
+
+    /**
+     * Gets image height and width for the bitstream. For jp2 images, height and width are
+     * obtained from the IIIF image server. For other formats supported by ImageIO these values
+     * are read from the actual DSpace bitstream content. If bitstream width metadata already exists,
+     * the bitstream is processed when forceProcessing is true.
+     * @param context
+     * @param bitstream
+     * @return
+     * @throws Exception
+     */
+    private boolean processBitstream(Context context, Bitstream bitstream) throws Exception {
+        boolean processed = false;
+        boolean isUnsupported = bitstream.getFormat(context).getMIMEType().contains("image/jp2");
+        boolean isImage = bitstream.getFormat(context).getMIMEType().contains("image/");
+        if (isImage) {
+            Optional<MetadataValue> op = bitstream.getMetadata().stream()
+                                                  .filter(m -> m.getMetadataField().toString('.')
+                                                                .contentEquals(IIIF_WIDTH_METADATA)).findFirst();
+            if (op.isEmpty() || forceProcessing) {
+                if (forceProcessing && !isQuiet) {
+                    System.out.println("Force processing for bitstream: " + bitstream.getID());
+                }
+                int[] dims;
+                if (isUnsupported) {
+                    dims = iiifApiQuery.getImageDimensions(bitstream);
+                } else {
+                    InputStream stream = bitstreamService.retrieve(context, bitstream);
+                    dims = ImageDimensionReader.getImageDimensions(stream);
+                }
+                if (dims != null) {
+                    processed = setBitstreamMetadata(context, bitstream, dims);
+                    // update the bitstream
+                    bitstreamService.update(context, bitstream);
+                }
+            }
+        }
+        return processed;
+    }
+
+    /**
+     * Sets bitstream metadata for "iiif.image.width" and "iiif.image.height".
+     * @param context
+     * @param bitstream
+     * @param dims
+     * @return
+     */
+    private boolean setBitstreamMetadata(Context context, Bitstream bitstream, int[] dims) throws Exception {
+        dSpaceObjectService.clearMetadata(context, bitstream, "iiif",
+                "image", "width", Item.ANY);
+        dSpaceObjectService.setMetadataSingleValue(context, bitstream, "iiif",
+                "image", "width", Item.ANY, String.valueOf(dims[0]));
+        dSpaceObjectService.clearMetadata(context, bitstream, "iiif",
+                "image", "height", Item.ANY);
+        dSpaceObjectService.setMetadataSingleValue(context, bitstream, "iiif",
+                "image", "height", Item.ANY, String.valueOf(dims[1]));
```

Comment:
Hi @abollini , so `IIIFUtils` and `ImageContentService` are part of the spring boot web application. Since we already have an `IIIFSharedUtils` class in `org.dspace.iiif` it might make sense for both the cli process and web application to use the shared utils for constants and the `info.json` path. I'll refactor accordingly. Let me know if you see a better approach.